### ISSUE TITLE
Checkbox style

### ISSUE THREAD

**Issue by bcholmes** [Hopding/pdf-lib#982]
When I create checkboxes in Acrobat Pro, I have options of style: check, cross, circle, etc. Can I specify these options in pdf-lib? Can I change the style on an existing checkbox, or even create a checkbox with a different style?

**Comment by Hopding** [comment]
Hello @bcholmes!

You can't specify the specific options that acrobat offers. That would be a great feature to offer though if you'd like to build a PR for it 😄 

However, you can specify a custom style for checkboxes (or any other form field). I build the acroforms API specifically to allow this. This integration test showcases the feature: https://github.com/Hopding/pdf-lib/blob/9168383d88e60b77922d2d031f4f9138eeff7240/apps/node/tests/test17.ts#L78-L194

You can run the test and view the resulting document by cloning the repo and running:
```
yarn install
yarn apps:node 17
```

**Comment by bcholmes** [comment]
Thanks for answering my question. 

I'd love to do a PR, once I have a bit more fluency with pdf-lib under my belt.

**Comment by marinoandrea** [comment]
Hi!

I would like to work on this. However, I was wondering what should the API to choose from a set of predefined checkbox appearances look like.

@Hopding, should some new `AppearanceProviderFor<PDFCheckBox>` consts simply be exported from the library (maybe in a submodule) in order to be directly passed to `PDFCheckBox.updateAppearances`?
Or maybe the default checkbox appearance should include the logic to optionally (e.g. based on a new argument) draw different shapes for the 'on' state with operations like `drawCheckMark`?

I still need to familiarize with this codebase but I stumbled upon this issue while working on a personal project and I think it would be a nice addition.

**Comment by freddy-pieters** [comment]
I had the same need to have a checkbox with the Acrobat "Cross" style and it took me a while to find an approach that works for me.

The problem with updating the appearances of a checkbox using pdf-lib is that Acrobat partly ignores them. I was successful in showing a cross when checked, but when clicking somewhere else on the page the appearance reverted to the "Check" style (the default). I could also see that in the properties of the checkbox.

Important is that the style of a checkbox is not part of the PDF specification. In other words, the style that Acrobat shows in the UI must be Acrobat-specific, but there is no custom entry in the PDFDict of the checkbox. The only explanation I see is that Acrobat checks the stream of the appearance to determine the style; by the way, the cross in a checkbox created by Acrobat is not made up of two lines, but of a character from the Zapf Dingbats font.

If that assumption is correct, then the problem must be solvable by making sure the whole PDFDict of the indirect object that describes the checkbox is identical to that as created by Acrobat.

I tried doing that, but I could not get it to work and I was also worried that a small detail could make the difference between success and frustration.

I then took the approach of duplicating the whole checkbox construct from a template file created with Acrobat and that does the trick. Attached is the code that can be run from the command line (as .ts files are not allowed I added .txt, so remove that and transpile).

[acrobatcheckbox.ts.txt](https://github.com/Hopding/pdf-lib/files/9806649/acrobatcheckbox.ts.txt)

The approach of using a template file also has its advantages, or at least it has for my use case: the user can edit the template file to create the checkbox of their liking and the code still works. I have not tried it, but it may also be a useful approach for the more complex widgets, e.g. text fields with a specific formatting, validation rules or calculated values.
